mars/services/task/execution/ray/executor.py

```diff
@@ -0,0 +1,261 @@
+# Copyright 1999-2021 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import logging
+from typing import List, Dict, Any, Set
+from .....core import ChunkGraph, Chunk
+from .....core.operand import (
+    Fuse,
+    VirtualOperand,
+    MapReduceOperand,
+    execute,
+)
+from .....lib.aio import alru_cache
+from .....optimization.physical import optimize
+from .....resource import Resource
+from .....serialization import serialize, deserialize
+from .....typing import BandType, TileableType
+from .....utils import (
+    lazy_import,
+    get_chunk_params,
+    get_chunk_key_to_data_keys,
+    ensure_coverage,
+)
+from ....cluster.api import ClusterAPI
+from ....lifecycle.api import LifecycleAPI
+from ....meta.api import MetaAPI
+from ....subtask import SubtaskGraph
+from ....subtask.utils import iter_input_data_keys, iter_output_data
+from ..api import TaskExecutor, ExecutionChunkResult, register_executor_cls
+
+ray = lazy_import("ray")
+logger = logging.getLogger(__name__)
+
+
+def execute_subtask_chunk_graph(
+    subtask_id: str, chunk_graph, output_meta_keys: Set[str], keys: List[str], *inputs
+):
```

Comment:
Maybe rename it to `execute_subtask` and should add type hint for the argument `chunk_graph`

Reply:
Good suggestion! execute_subtask is better.